pkg/monitor/dynkeepalived.go

```diff
@@ -87,20 +87,22 @@ func doesConfigChanged(curConfig, appliedConfig *config.Node) bool {
 }
 
 func retrieveBootstrapIpAddr(apiVip string) {
-	var err error
 
-	if gBootstrapIP != "" {
-		return
-	}
 	// we don't need to read the bootstrap IP address for bootstrap node
 	if os.Getenv("IS_BOOTSTRAP") == "yes" {
 		gBootstrapIP = ""
 		return
 	}
-	gBootstrapIP, err = config.GetBootstrapIP(apiVip)
-	if err != nil {
-		log.Debugf("Could not retrieve bootstrap IP: %v", err)
+
+	retrievedIP, err := config.GetBootstrapIP(apiVip)
+	if gBootstrapIP != retrievedIP {
```

Comment:
Feels a little weird to be using the return value when err != nil, but I guess it's fine since GetBootstrapIP returns "" on error.